src/Algebra/Graph.hs

```diff
@@ -1051,6 +1153,8 @@ induceJust = foldg Empty (maybe Empty Vertex) (k Overlay) (k Connect)
 -- Complexity: the function performs /O(s)/ graph comparisons. It is guaranteed
 -- that the size of the result does not exceed the size of the given expression.
 --
+-- Good consumer.
```

Comment:
Let's make it a good producer too (seems straightforward).

Reply:
This is not straightforward: when you abstract your constructors using `buildg` you loose the `Eq` instance required by `simple` (intuitively, you could destruct your graph to a value which does not have an `Eq` instance)

Reply:
Hmm, I see, that's interesting. Worth documenting this intuition!